src/azure-cli/azure/cli/command_modules/resource/_params.py

```diff
@@ -55,6 +55,12 @@ def load_arguments(self, _):
     no_prompt = CLIArgumentType(arg_type=get_three_state_flag(), help='The option to disable the prompt of missing parameters for ARM template. '
                                 'When the value is true, the prompt requiring users to provide missing parameter will be ignored. The default value is false.')
 
+    deployment_what_if_result_format_type = get_enum_type(WhatIfResultFormat, WhatIfResultFormat.full_resource_payloads)
```

Comment:
May I know what is the difference for these two formats?

Reply:
With `full_resource_payloads` the What-If API returns detailed resource changes with the resource ID, before state, after state and delta information calculated from the before and after states. With `resource_id_only` the API only return resource changes with resource ID.